huaweicloud/services/dcs/data_source_huaweicloud_dcs_instances.go

```diff
@@ -0,0 +1,364 @@
+			},
+			"access_user": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the username used for accessing a DCS Memcached instance.`,
+			},
+			"order_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the ID of the order that created the instance.`,
+			},
+			"tags": common.TagsSchema(),
+		},
+	}
+	return &sc
+}
+
+func resourceDcsInstanceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getDCSInstances: Query the List of DCS instances.
+	var (
+		getDCSInstancesHttpUrl = "v2/{project_id}/instances"
+		getDCSInstancesProduct = "dcs"
+	)
+	getDCSInstancesClient, err := config.NewServiceClient(getDCSInstancesProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DcsInstance Client: %s", err)
+	}
+
+	getDCSInstancesPath := getDCSInstancesClient.Endpoint + getDCSInstancesHttpUrl
+	getDCSInstancesPath = strings.ReplaceAll(getDCSInstancesPath, "{project_id}", getDCSInstancesClient.ProjectID)
+
+	getDCSInstancesQueryParams := buildGetDCSInstancesQueryParams(d)
+	getDCSInstancesPath += getDCSInstancesQueryParams
+
+	getDCSInstancesResp, err := pagination.ListAllItems(
+		getDCSInstancesClient,
+		"offset",
+		getDCSInstancesPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving DcsInstance")
+	}
+
+	getDCSInstancesRespJson, err := json.Marshal(getDCSInstancesResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var getDCSInstancesRespBody interface{}
+	err = json.Unmarshal(getDCSInstancesRespJson, &getDCSInstancesRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instances", flattenGetDCSInstancesResponseBodyInstance(getDCSInstancesRespBody, getDCSInstancesClient)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenGetDCSInstancesResponseBodyInstance(resp interface{}, client *golangsdk.ServiceClient) []interface{} {
+	if resp == nil {
+		return nil
+	}
+	curJson := utils.PathSearch("instances", resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+	rst := make([]interface{}, 0, len(curArray))
+	for _, v := range curArray {
+		// capacity
+		capacity := utils.PathSearch("capacity", v, nil)
+		capacityMinor := utils.PathSearch("capacity", v, nil)
+		if capacity == 0 {
+			capacity, _ = strconv.ParseFloat(capacityMinor.(string), floatBitSize)
+		}
+
+		securityGroupID := utils.PathSearch("security_group_id", v, nil)
+		// If security_group_id is not set, the default value is returned: securityGroupId. Bad design.
+		if securityGroupID == "securityGroupId" {
+			securityGroupID = ""
+		}
```

Comment:
the description "If security_group_id is not set, the default value is returned: securityGroupId. Bad design" not match the code: 
if securityGroupID == "securityGroupId" {
			securityGroupID = ""
}
please check it.